src/uct/ib/dc/accel/dc_mlx5.c

```diff
@@ -571,11 +576,35 @@ ucs_status_t uct_dc_mlx5_ep_fc_ctrl(uct_ep_t *tl_ep, unsigned op,
     UCT_DC_CHECK_RES(&iface->super, dc_ep);
     UCT_DC_MLX5_IFACE_TXQP_GET(iface, dc_ep, txqp, txwq);
 
+    dc_req = ucs_derived_of(req, uct_dc_fc_request_t);
+
     if (op == UCT_RC_EP_FC_PURE_GRANT) {
         ucs_assert(req != NULL);
-        dc_req    = ucs_derived_of(req, uct_dc_fc_request_t);
-        sender_ep = (uintptr_t)dc_req->sender_ep;
+
         ib_iface  = &iface->super.super.super;
+        sender_ep = (uintptr_t)dc_req->sender.ep;
+
+        if ((ib_iface->addr_type == UCT_IB_ADDRESS_TYPE_GLOBAL) ||
+            (dc_req->sender.gid.global.subnet_prefix != ib_iface->gid.global.subnet_prefix)) {
+            ah_attr.sl             = ib_iface->config.sl;
+            ah_attr.src_path_bits  = ib_iface->path_bits[0];
+            ah_attr.dlid           = dc_req->lid | ib_iface->path_bits[0];
```

Comment:
Can this code be united with uct_ud_mlx5_iface_get_av somehow?

Reply:
there is bit different interface, will require additional refactoring for uct_ud_mlx5_iface_get_av. will add additional PR later